### PR TITLE
Cherrypick #5353 into release-v5.2

### PR DESCRIPTION
**Pull request by Amxx**
Will get into RC.1

### CODE REVIEW

**Comment by Amxx** [comment]
DO not merge until #5371 is merged an cherrypicked

**Review by arr00 — APPROVED** [review]

**Comment by Amxx** [comment]
Since we changing an unreleased feature, and there is no Changelog difference between the RC (this is not a patch), I don't think we should have a changelog entry for this. 

We'll fix that when we cleanup the changelog, otherwise it will mess up the list of changeset between here and master and it risk showing up again.

**Pull request merged**
